pandas/tests/indexing/test_loc.py

```diff
@@ -3016,7 +3017,15 @@ def test_loc_setitem_uint8_upcast(value):
     with tm.assert_produces_warning(FutureWarning, match="item of incompatible dtype"):
         df.loc[2, "col1"] = value  # value that can't be held in uint8
 
-    expected = DataFrame([1, 2, 300, 4], columns=["col1"], dtype="uint16")
+    if np_version_gt2 and isinstance(value, np.int16):
+        # Note, result type of uint16 + int16 is int16
```

Comment:
Maybe the ship has sailed on this but am I reading this right that `uint16` + `int16` in numpy > 2 yields an `int16`? 

Just surprising to me as that is not how usual arithmetic conversions happen in C

Reply:
Oh, good catch, I think it should read uint8 + int16 = int16.
I think I typo'ed somewhere.

In hindsight, I think I might have also "xfailed" this test, when I shouldn't have.
I think in this case we'd want to preserve the unsigned datatype instead of trying to see if an int would work.

Reply:
Ah ok great. That makes more sense

Reply:
I think I was wrong again in the previous comment 😆, about the behavior being wrong. 

I thought this test went through code that I changed, but instead it basically just goes through np.result_type directly, like previously.

Reply:
So uint16 + int16 should yield int16? That seems very strange to me - what is the expected output of `2 ** 15 + 1`?

Reply:
No, uint8+int16=int16.

The second part of my comment was wrong.

Reply:
I fixed the typo in the latest commit I think.

Reply:
Ah ok - that makes a lot more sense. Thanks for clarifying